### PR TITLE
Current frontend progress

### PR DESCRIPTION
**Pull request by PierreBeur**

### CODE REVIEW

**Review by isabelle-kr — APPROVED** [review]

File: source/styles/main.css [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+main {
+  display: flex;
+  flex-grow: 1;
+  max-height: calc(100% - 4rem - 2px); /* HACK: subtract height of header */
+  padding: 1rem;
+}
+
+#job-details-list {
+  flex-grow: 1;
+  overflow-y: scroll;
```
Comment (JessalynWang) [inline source/styles/main.css:29]: Can we change this to auto so the scrollbar doesnt show up when there isnt enough data to scroll

File: source/reference/sort-bar.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<div>
+  <label for="field">
```
Comment (JessalynWang) [inline source/reference/sort-bar.html:2]: Not too sure if this is the right place to put it but can we fix the functionality on the sort end? To me it is unintuitive for the toggle sort and toggle reverse to be different things, as its confusing when you click the arrow only and the arrow moves but the scroll isnt toggled, I think a possible better interaction would be a three click toggle, where clicks change state (ex click one to sort normally, click two to sort reverse, click three to revert to not sorting state)

File: source/scripts/JobDetails.js [hunk 3]
```diff
@@ -68,47 +208,100 @@ class JobDetails extends HTMLElement {
...
-      </div>
-      <h1>Contact</h1>
-      <a href="${data.contact}">${data.contact}</a>
-      <h1>Portal</h1>
-      <a href="${data.portal}">${data.portal}</a>
-      <h1>Tags</h1>
-      <div id="tags">
+      <div class="dropdown">
+        <hr id="horizontal-rule">
+        <div class="heading">Description</div>
```
Comment (JessalynWang) [inline source/scripts/JobDetails.js:228]: add some ternaries for some default text that shows up if fields are empty

**Review by JessalynWang — CHANGES_REQUESTED** [review]

File: source/styles/main.css (hunk 1 above)
Comment (PierreBeur) [reply source/styles/main.css:29]: Unfortunately this screws up the spacing between the job-details elements and the side of the screen when the scrollbar is not visible. I looked into fixing it with scrollbar styles but these are not featured enough to solve the issue, and are completely unsupported on Firefox. In my opinion, due to these issues, it is probably better for the scrollbar to stay visible at all times to provide consistent spacing.

image

image

image

image

File: source/scripts/JobDetails.js (hunk 3 above)
Comment (PierreBeur) [reply source/scripts/JobDetails.js:228]: Would it be better for the heading for that field to not show up when the field is empty?
Here is an before/after example for an empty "Notes" field:

image

image

As opposed to:

image

File: source/reference/sort-bar.html (hunk 2 above)
Comment (PierreBeur) [reply source/reference/sort-bar.html]: Yeah, good idea. Made the changes and it works as described:
image
image
image
image

File: source/styles/main.css (hunk 1 above)
Comment (JessalynWang) [reply source/styles/main.css:29]: have you tried this? [https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter)

Comment (PierreBeur) [reply source/styles/main.css]: Thanks for the suggestion! Seems to have worked. The spacing is now consistent whether or not the scrollbar is visible.

image

image

File: source/scripts/JobDetails.js (hunk 3 above)
Comment (PierreBeur) [reply source/scripts/JobDetails.js]: Just added the change:

image

**Review by JessalynWang — APPROVED** [review]

**Pull request merged**